server/api/insights.go

```diff
@@ -113,7 +113,12 @@ func (a *API) handleTeamBoardsInsights(w http.ResponseWriter, r *http.Request) {
 		userLocation = time.Now().UTC().Location()
 	}
 	// get unix time for duration
-	startTime := mmModel.StartOfDayForTimeRange(timeRange, userLocation)
+	startTime, appErr := mmModel.StartOfDayForTimeRange(timeRange, userLocation)
+	if appErr != nil {
+		a.errorResponse(w, r, appErr)
```

Comment:
Can we create a new error with `ErrBadRequest` with just message from `appErr`? Same for the changes in user boards insights?